src/components/vm/overview/watchdog.jsx

```diff
@@ -0,0 +1,138 @@
+    Modal,
+    Radio,
+} from '@patternfly/react-core';
+import { useDialogs } from 'dialogs.jsx';
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { domainRemoveWatchdog, domainSetWatchdog } from "../../../libvirtApi/domain.js";
+import { rephraseUI } from "../../../helpers.js";
+import { WarningInactiveAlert, WarningInactiveTooltip } from "../../common/warningInactive.jsx";
+
+const _ = cockpit.gettext;
+
+export const WatchdogModal = ({ vm, isWatchdogAttached, idPrefix }) => {
+    const [dialogError, setDialogError] = useState();
+    const [watchdogAction, setWatchdogAction] = useState(vm.watchdog.action || "no-device");
+
+    const Dialogs = useDialogs();
+
+    const save = () => domainSetWatchdog({
+        connectionName: vm.connectionName,
+        vmName: vm.name,
+        hotplug: vm.state === "running",
+        permanent: vm.persistent,
+        action: watchdogAction,
+        isWatchdogAttached,
+    })
+            .then(Dialogs.close, exc => setDialogError({ text: _("Failed to configure watchdog"), detail: exc.message }));
+
+    const detach = () => domainRemoveWatchdog({
+        connectionName: vm.connectionName,
+        vmName: vm.name,
+        hotplug: vm.state === "running",
+        permanent: vm.persistent,
+        model: vm.watchdog.model,
+    })
+            .then(Dialogs.close, exc => setDialogError({ text: _("Failed to detach watchdog"), detail: exc.message }));
+
+    const supportedActions = ["reset", "poweroff", "inject-nmi", "pause"];
+
+    const showWarning = () => {
+        if (vm.state === 'running' && isWatchdogAttached && vm.watchdog.action !== watchdogAction)
+            return <WarningInactiveAlert idPrefix={idPrefix} />;
+    };
+
+    return (
+        <Modal id={`${idPrefix}-watchdog-modal`}
+               position="top"
+               variant="small"
+               onClose={Dialogs.close}
+               title={_("Watchdog device type")}
+               description={_("Watchdogs act when systems stop responding. To use this virtual watchdog device, the guest system also needs to have a running watchdog service.")}
+               isOpen
+               footer={
+                   <>
+                       {dialogError && <ModalError dialogError={dialogError.text} dialogErrorDetail={dialogError.detail} />}
+                       <Button variant='primary' id="watchdog-dialog-apply" onClick={save}>
+                           {isWatchdogAttached ? _("Save") : _("Add")}
+                       </Button>
+                       {isWatchdogAttached && <Button variant='secondary' id="watchdog-dialog-detach" onClick={detach}>
+                           {_("Remove")}
+                       </Button>}
+                       <Button variant='link' onClick={Dialogs.close}>
+                           {_("Cancel")}
+                       </Button>
+                   </>
+               }>
+            {showWarning()}
+            <Form isHorizontal>
```

Comment:
Add also onSubmit={e => e.preventDefault()}  to avoid page reloading when hitting enter.